### PR TITLE
[upcoming] UIE-9410: Implement the Create Share Group page

### PR DESCRIPTION
**Pull request by fabrice-akamai**
## Description 📝

This PR implements the 'Create Share Group' page along with the supporting functions for querying, routing, and submitting the form. With these changes, users can now create share groups from the UI instead of relying on API requests alone. This is the first of 3 PRs for this ticket; future PRs will implement the Images and Selected images sections. 

## Changes  🔄

List any change(s) relevant to the reviewer.

- Added ShareGroupsCreate page with landing header and form
- Added ShareGroupsCreate form
- Added ShareGroupsCreate query mutation in `sharegroups.ts`
- Added new route in `routes/images/index.ts`
- Added unit tests for the ShareGroupsCreate form

### Scope 🚢

 Upon production release, changes in this PR will be visible to:

- [ ] All customers
- [x] Some customers (e.g. in Beta or Limited Availability)
- [ ] No customers / Not applicable

## Target release date 🗓️

TBD

## Preview 📷

https://github.com/user-attachments/assets/a2dca6dc-dde9-40eb-8723-f4a37d773cf3

### Prerequisites

In an environment with the Private Image Sharing feature flag enabled,

- Navigate to Images -> Share Groups tab
- Click the 'Create Share Group' button on the landing page

### Verification steps

(How to verify changes)

- [ ] Fill the label and optionally the description field
- [ ] Click Create Share Group
- [ ] Verify that your share group was successfully created and appears in the table 
- [ ] Test edge cases such as submitting an empty form, missing label, spaces and special characters in label, exceeding character limit in label field, etc. Make sure the form displays the correct error message and handles errors gracefully
- [ ] Verify that the breadcrumb and docs links work properly

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>

- [ ] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [ ] All tests and CI checks are passing
- [ ] TypeScript compilation succeeded without errors
- [ ] Code passes all linting rules

</details>

### CODE REVIEW

File: packages/manager/.changeset/pr-13550-upcoming-features-1774974481770.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@linode/manager": Upcoming Features
+---
+
+Implement the basic share group create page ([#13550](https://github.com/linode/manager/pull/13550))
```
Comment (dwiley-akamai) [inline packages/manager/.changeset/pr-13550-upcoming-features-1774974481770.md]:
```suggestion
Private Image Sharing: Implement basic structure of Share Group Create page ([#13550](https://github.com/linode/manager/pull/13550))
```

File: packages/queries/src/images/sharegroups.ts [hunk 2]
```diff
@@ -95,3 +102,21 @@ export const useShareGroupsInfiniteQuery = (
...
+
+export const useCreateShareGroupMutation = () => {
+  const queryclient = useQueryClient();
+
+  return useMutation<Sharegroup, APIError[], CreateSharegroupPayload>({
+    mutationFn: createSharegroup,
+    onSuccess(shareGroup) {
+      queryclient.invalidateQueries({
+        queryKey: shareGroupsQueries.sharegroups._ctx.paginated._def,
+      });
```
Comment (dwiley-akamai) [inline packages/queries/src/images/sharegroups.ts:114]: I think we'll also want

```
      queryclient.invalidateQueries({
        queryKey: shareGroupsQueries.sharegroups._ctx.all._def,
      });

      queryclient.invalidateQueries({
        queryKey: shareGroupsQueries.sharegroups._ctx.infinite._def,
      });
```

File: packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreate.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                onChange={(e) =>
+                  field.onChange(
+                    e.target.value === '' ? undefined : e.target.value
+                  )
+                }
+                rows={1}
+                value={field.value ?? ''}
+              />
+            )}
+          />
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreate.tsx:93]: Let's check with the API team on implementing a maximum number of characters for the `description` field

Image

File: packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreate.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+  const selectedImages = [];
+
+  const onSubmit = handleSubmit(async (values) => {
+    try {
+      await createShareGroup(values);
+
+      enqueueSnackbar('Sharegroup scheduled for creation', {
+        variant: 'info',
+      });
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreate.tsx]: Was a snackbar in the wireframes? I think the user will just be directed to the Share Group detail page upon successful creation. Or is this just temporary until that's implemented? (we should have a code comment in that case)

Comment (fabrice-akamai) [reply packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreate.tsx]: I meant to remove it, thanks for catching this. I realize we don't need a snackbar here as the user will just be directed to the Share Group landing page.

File: packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreate.tsx (hunk 3 above)
Comment (fabrice-akamai) [reply packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreate.tsx:93]: Good point, I'll reach out to them about this.

File: packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreateContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import Grid from '@mui/material/Grid';
```
Comment (abailly-akamai) [inline packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreateContainer.tsx:1]: Let's no do straight imports from `@mui`

**Review by abailly-akamai — APPROVED** [review]
Nice - PR looks good from a code/UI perspective 👍

File: packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreateContainer.tsx (hunk 5 above)
Comment (dwiley-akamai) [reply packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreateContainer.tsx:1]: Is there an alternative you suggest @abailly-akamai? We do this throughout the codebase and I don't believe we have a `Grid` in the `ui` package

**Review by dwiley-akamai — APPROVED** [review]
Verification steps ✅
Code review ✅

File: packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreateContainer.tsx (hunk 5 above)
Comment (abailly-akamai) [reply packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreateContainer.tsx:1]: 🤕 My bad old habit when I see a MUI import. It's one of the very few component we haven't abstracted. Carry on!

oof-the-simpsons

Comment (fabrice-akamai) [reply packages/manager/src/features/Images/ImagesLanding/v2/ShareGroups/ShareGroupsCreate/ShareGroupsCreateContainer.tsx:1]: Haha, no worries. Thanks for clarifying this @dwiley-akamai @abailly-akamai!

**Comment by fabrice-akamai** [comment]
I made a few additional changes to the ShareGroupsTable to improve the display of the Group and Description fields when the text is too large: 

- Removed line wrap so that the text overflow gets truncated with ellipses
- Added a tooltip to show the full text 
- Made the description column wider 

| Same width (current) | Wider description column (new) |
|--------|--------|
| Screenshot 2026-04-01 at 6 36 16 PM | Screenshot 2026-04-01 at 6 50 27 PM |

@tzmiivsk-akamai any thoughts on expanding the description column and adding a tooltip to the group cells? 

cc: @dwiley-akamai @abailly-akamai

**Comment by tzmiivsk-akamai** [comment]
> I made a few additional changes to the ShareGroupsTable to improve the display of the Group and Description fields when the text is too large:
> 
> * Removed line wrap so that the text overflow gets truncated with ellipses
> * Added a tooltip to show the full text
> * Made the description column wider
> 
> @tzmiivsk-akamai any thoughts on expanding the description column and adding a tooltip to the group cells?

Looking good! A couple of small adjustments:
Let's show tooltips only when the description is truncated to avoid visual noise.
Can we narrow down 'Members'/'Images' columns and expand 'Group' instead?
Paddings (left of 'Group' and right of 'Action Menu') look a bit wider than the designs.

Could you please double-check these? Thanks!

**Comment by fabrice-akamai** [comment]
Latest changes: 

- Updated the tooltip in the `group` and `description` to show only when the text gets truncated for exceeding the character limit (32 for group and 50 for description)
- Implemented different column widths depending on screen size and column type


https://github.com/user-attachments/assets/a30b7a1f-9e59-4218-8415-cab16cb9f406



Screenshot 2026-04-06 at 3 46 56 PM

**Comment by fabrice-akamai** [comment]
> > I made a few additional changes to the ShareGroupsTable to improve the display of the Group and Description fields when the text is too large:
> > 
> > * Removed line wrap so that the text overflow gets truncated with ellipses
> > * Added a tooltip to show the full text
> > * Made the description column wider
> > 
> > @tzmiivsk-akamai any thoughts on expanding the description column and adding a tooltip to the group cells?
> 
> Looking good! A couple of small adjustments: Let's show tooltips only when the description is truncated to avoid visual noise. Can we narrow down 'Members'/'Images' columns and expand 'Group' instead? Paddings (left of 'Group' and right of 'Action Menu') look a bit wider than the designs.
> 
> Could you please double-check these? Thanks!

@tzmiivsk-akamai I wasn't able to remove the horizontal padding along the table row because it's built into the CDS table we're using, and it doesn't allow us to override the default padding there. I tried a few different solutions, but none worked. I will report this issue to the CDS team so it can be addressed in the future.
